### ISSUE TITLE
go-yaml v1.13.7 panic on the the input JSON string for the map slice type.

### ISSUE THREAD

**Issue by Peefy** [goccy/go-yaml#519]
**Describe the bug**

**To Reproduce**

Please provide a minimum yaml content that can be reproduced.
We are more than happy to use [Go Playground](https://go.dev/play)

Use the go 1.23 and `github.com/goccy/go-yaml v1.13.7` to run the following test, it will panic with the error message but `github.com/goccy/go-yaml` `v1.13.4`, `v1.13.5` and `v1.13.6` work fine

```go
func TestPanicJsonYamlMapSlice(t *testing.T) {
	code := `
{
	"apiVersion": "apps/v1",
	"kind": "Deployment",
	"metadata": {
		"name": "nginx-deployment",
		"labels": {
			"app": "nginx"
		}
	},
	"spec": {
		"replicas": 3,
		"selector": {
			"matchLabels": {
				"app": "nginx"
			}
		},
		"template": {
			"metadata": {
				"labels": {
					"app": "nginx"
				}
			},
			"spec": {
				"containers": [
					{
						"name": "nginx-container",
						"image": "nginx:latest",
						"ports": [
							{
								"containerPort": 80
							}
						]
					}
				]
			}
		}
	}
}	
`
	yamlData := &yaml.MapSlice{}
	if err := yaml.UnmarshalWithOptions([]byte(code), yamlData, yaml.UseOrderedMap(), yaml.UseJSONUnmarshaler()); err != nil {
		panic(err)
	}
}
```

The error message is

```txt
panic: [10:3] value is not allowed in this context
           7 |          "labels": {
           8 |                  "app": "nginx"
           9 |          }
        > 10 |  },
                 ^
          11 |  "spec": {
          12 |          "replicas": 3,
          13 |          "selector": { [recovered]
        panic: [10:3] value is not allowed in this context
           7 |          "labels": {
           8 |                  "app": "nginx"
           9 |          }
        > 10 |  },
                 ^
          11 |  "spec": {
          12 |          "replicas": 3,
          13 |          "selector": {
```

**Expected behavior**

Test passed

**Screenshots**

**Version Variables**
 - Go version: [e.g. 1.21 ] 1.23
 - go-yaml's Version: [e.g. v1.11.1 ] v1.13.7

**Additional context**

**Comment by yxxhero** [comment]
@Peefy I found same issue: https://github.com/helmfile/helmfile/actions/runs/11781045144/job/32813652793?pr=1779

**Comment by goccy** [comment]
Thank you for your reporting! I'll fix this problem today

**Comment by goccy** [comment]
I've fixed this problem with [v1.13.8](https://github.com/goccy/go-yaml/releases/tag/v1.13.8) . Please confirm it 🙏
